core/src/main/java/com/scalar/db/storage/dynamo/DynamoAdmin.java

```diff
@@ -743,9 +778,55 @@ private void waitForTableDeletion(Namespace namespace, String tableName)
 
   @Override
   public void dropNamespace(String nonPrefixedNamespace) throws ExecutionException {
-    Set<String> tables = getNamespaceTableNames(nonPrefixedNamespace);
+    Namespace namespace = Namespace.of(namespacePrefix, nonPrefixedNamespace);
+    try {
+      deleteFromNamespaceTable(namespace);
+      dropAllTablesOfNamespace(namespace);
+      dropNamespaceTableIfEmpty();
+    } catch (Exception e) {
+      throw new ExecutionException(
+          "dropping the namespace "
+              + Namespace.of(namespacePrefix, nonPrefixedNamespace)
+              + " failed",
+          e);
```

Comment:
It looks like this can throw a `ExecutionException` whose `cause` is also `ExecutionException`?

Reply:
Yes, I do this to give more context to the `ExecutionException` thrown by a sub-method. I think the stack trace produced makes it easier to understand the root reason for the error.